src/main/java/org/dependencytrack/persistence/QueryManager.java

```diff
@@ -314,6 +315,22 @@ private CacheQueryManager getCacheQueryManager() {
         return cacheQueryManager;
     }
 
+    /**
+     * Disables the second level cache for this {@link QueryManager} instance.
+     * <p>
+     * Disabling the L2 cache is useful in situations where large amounts of objects
+     * are created or updated in close succession, and it's unlikely that they'll be
+     * accessed again anytime soon. Keeping those objects in cache would unnecessarily
+     * blow up heap usage.
+     *
+     * @return This {@link QueryManager} instance
+     * @see <a href="https://www.datanucleus.org/products/accessplatform_6_0/jdo/persistence.html#cache_level2">L2 Cache docs</a>
+     */
+    public QueryManager withL2CacheDisabled() {
+        pm.setProperty(PropertyNames.PROPERTY_CACHE_L2_TYPE, "none");
+        return this;
+    }
+
```

Comment:
With this method on the `QueryManager` I would expect the L2 Cache to be enabled again on subsequent quermanagers created from the same `pm`. Is this the case? It looks to me as if L2 is disabled for the life of the `pm` and not the `QueryManager`? In this case where the whole tasks is supposed to have L2 disabled it makes no difference, but it could be confusing for other scenario's where only a couple of queries want to have their L2 disabled?

Reply:
> With this method on the QueryManager I would expect the L2 Cache to be enabled again on subsequent quermanagers created from the same pm. Is this the case? It looks to me as if L2 is disabled for the life of the pm and not the QueryManager?

Yes, the setting applies to the lifetime of the `PersistenceManager`. Every `QueryManager` creates a new underlying `PersistenceManager`, and closes it again when it itself is closed. So the lifetime of `QueryManager` and `PersistenceManager` is tightly coupled.

> but it could be confusing for other scenario's where only a couple of queries want to have their L2 disabled?

Hmmm... Any suggestions on making this more obvious? AFAICT for individual queries one would use

```java
var query = pm.newQuery(...);
query.setIgnoreCache(true);
```

And for writing operations one *must* use the `PersistenceManager` setting.